src/core/shape_inference/include/pooling_shape_inference_util.hpp

```diff
@@ -151,6 +157,40 @@ void valid_dilated_kernel_with_padding(const TOp* op,
                                        const size_t pad_end,
                                        const size_t axis) {}
 
+template <class TDim>
+void align_ceil_torch_dimension_size(TDim& dim,
+                                     const size_t last_pooling_start_index,
+                                     const size_t data_dim_length,
+                                     const size_t pads_begin) {
+    if (!(last_pooling_start_index > data_dim_length + pads_begin - 1)) {
+        dim += 1;
+    }
+}
+
+template <class TDim>
+TDim disallow_pooling_start_in_padding(TDim& dim, const size_t stride, const TDim* data_dim, const size_t pads_begin) {
+    // Ensure the last pooling doesn't start in padding.
+    auto dim_min_length = dim.get_min_length();
+    const auto last_pooling_min_start_index = (dim_min_length - 1) * stride;
+    const auto data_dim_min_length = data_dim->get_min_length();
+    align_ceil_torch_dimension_size(dim_min_length, last_pooling_min_start_index, data_dim_min_length, pads_begin);
+    if (data_dim->is_static()) {
+        return TDim(dim_min_length);
+    } else {
+        auto dim_max_length = dim.get_max_length();
+        const auto last_pooling_max_start_index = (dim_max_length - 1) * stride;
+        const auto data_dim_max_length = data_dim->get_max_length();
+        align_ceil_torch_dimension_size(dim_max_length, last_pooling_max_start_index, data_dim_max_length, pads_begin);
+        return TDim(dim_min_length, dim_max_length);
+    }
+}
+
+template <class TDim>
+TDim allow_pooling_start_in_padding(TDim& dim, const size_t, const TDim*, const size_t) {
+    dim += 1;
+    return dim;
+}
```

Comment:
```suggestion
TDim allow_pooling_start_in_padding(const TDim& dim, const size_t, const TDim*, const size_t) {
    return dim + 1;
}
```
As new dimension is create the input value should not be modified.